Update rustdoc.css

BradMarr opened the pull request:
fixes #120595


removing this selector fixes weird wrapping in certain pages like https://docs.rs/async-stripe/0.31.0/stripe/index.html#structs

clubby789 committed:
Update `icu4x` dependencies

clubby789 committed:
Update `r-efi`

clubby789 committed:
Run `cargo update`

jieyouxu committed:
Autolabel run-make tests, remind to update tracking issue

- Autolabel PRs modifying `tests/run-make/` and
  `src/tools/run-make-support/` with `X-run-make` label.
- Add reminder to update the tracking issue
  <https://github.com/rust-lang/rust/issues/121876>
  if applicable when `tests/run-make/` is modified by a PR.

ChrisDenton committed:
Migrate more things to WinError

notriddle committed:
rustdoc-search: use lowercase, non-normalized name for type search

The type name ID map has underscores in its names, so the query
element should have them, too.

BradMarr committed:
Update rustdoc.css

GuillaumeGomez commented:
This indeed seems to fix the issue, thanks!

Please add a test in `tests/rustdoc-gui` to ensure this bug won't appear again later on. If you need help to write the test, don't hesitate to ask!

BradMarr commented:
I'm not familiar with the test system and don't know I could prevent this issue from occurring in the future. The issue was simply that the property was spread to an element it shouldn't have been so I don't know if a test could detect a problem in the future without knowing the intention of the programmer.

rust-log-analyzer commented:
The job **`x86_64-gnu-tools`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/9439732430/job/25998192577) [(plain)](https://github.com/rust-lang/rust/commit/ee6459d6521cf6a4c2e08b6e13ce3c6ce5d55ed0/checks/25998192577/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
.......... (120/124)
...F       (124/124)


/checkout/tests/rustdoc-gui/type-declation-overflow.goml type-declation-overflow... FAILED
[ERROR] `tests/rustdoc-gui/type-declation-overflow.goml` line 19: Error: Evaluation failed: The following errors happened (for selector `.item-table .struct`): [expected `684` for property `"offsetWidth"`, found `126`]: for command `assert-property: (".item-table .struct", {"offsetWidth": "684"})`
Error: ()
Build completed unsuccessfully in 0:03:27
  local time: Sun Jun  9 22:38:37 UTC 2024
  network time: Sun, 09 Jun 2024 22:38:37 GMT
```

</details>

BradMarr commented:
I'm lost as to how an offset error would occur when my only change was to the overflow-wrap.

rust-log-analyzer commented:
The job **`x86_64-gnu-tools`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/9440409225/job/25999673032) [(plain)](https://github.com/rust-lang/rust/commit/44263c620269ef269f5eada5afaafb41b0288008/checks/25999673032/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
.......... (120/124)
...F       (124/124)


/checkout/tests/rustdoc-gui/type-declation-overflow.goml type-declation-overflow... FAILED
[ERROR] `tests/rustdoc-gui/type-declation-overflow.goml` line 19: Error: Evaluation failed: The following errors happened (for selector `.item-table .struct`): [expected `684` for property `"offsetWidth"`, found `126`]: for command `assert-property: (".item-table .struct", {"offsetWidth": "684"})`
Error: ()
Build completed unsuccessfully in 0:03:27
  local time: Mon Jun 10 00:46:29 UTC 2024
  network time: Mon, 10 Jun 2024 00:46:29 GMT
```

</details>

BradMarr commented:
maybe we could do a minimum width along with the default method so that it only makes the lines overlap in that way if the edge case occurs

BradMarr commented:
That would require parsing through all of the words, I think maybe giving each of the `.item-table>li>.item-name` a max width of say 600px it would allow for overflow of the itemname but only if it would otherwise cause a bug

BradMarr committed:
Update rustdoc.css

Update rustdoc.css

Update rustdoc.css

Update rustdoc.css

Update rustdoc.css

Update rustdoc.css

Update rustdoc.css

Update type-declation-overflow.goml

fix test

make into flex

BradMarr commented:
How does that commit look?

rust-log-analyzer commented:
The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/9451984156/job/26034193539) [(plain)](https://github.com/rust-lang/rust/commit/8bb8eab18bfd12f07fefeb1e58a225b031401fa9/checks/26034193539/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
Getting action download info
Download action repository 'msys2/setup-msys2@v2.22.0' (SHA:cc11e9188b693c2b100158c3322424c4cc1dadea)
Download action repository 'actions/checkout@v4' (SHA:a5ac7e51b41094c92402da3b24376905380afc29)
Download action repository 'actions/upload-artifact@v4' (SHA:65462800fd760344b1a7b4382951275a0abb4808)
Complete job name: PR - mingw-check-tidy
git config --global core.autocrlf false
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
---
COPY scripts/sccache.sh /scripts/
RUN sh /scripts/sccache.sh

COPY host-x86_64/mingw-check/reuse-requirements.txt /tmp/
RUN pip3 install --no-deps --no-cache-dir --require-hashes -r /tmp/reuse-requirements.txt \
    && pip3 install virtualenv
COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---

#12 [5/8] COPY host-x86_64/mingw-check/reuse-requirements.txt /tmp/
#12 DONE 0.0s

#13 [6/8] RUN pip3 install --no-deps --no-cache-dir --require-hashes -r /tmp/reuse-requirements.txt     && pip3 install virtualenv
#13 0.422   Downloading binaryornot-0.4.4-py2.py3-none-any.whl (9.0 kB)
#13 0.435 Collecting boolean-py==4.0
#13 0.439   Downloading boolean.py-4.0-py3-none-any.whl (25 kB)
#13 0.450 Collecting chardet==5.1.0
---
#13 3.489 Building wheels for collected packages: reuse
#13 3.490   Building wheel for reuse (pyproject.toml): started
#13 3.823   Building wheel for reuse (pyproject.toml): finished with status 'done'
#13 3.824   Created wheel for reuse: filename=reuse-1.1.0-cp310-cp310-manylinux_2_35_x86_64.whl size=181117 sha256=f5f58750481f69515c2c0d1d503daf565e2565c370d07fc6aeb95fe3498b4269
#13 3.824   Stored in directory: /tmp/pip-ephem-wheel-cache-2mat1pe_/wheels/c2/3c/b9/1120c2ab4bd82694f7e6f0537dc5b9a085c13e2c69a8d0c76d
#13 3.827 Installing collected packages: boolean-py, binaryornot, setuptools, reuse, python-debian, markupsafe, license-expression, jinja2, chardet
#13 3.849   Attempting uninstall: setuptools
#13 3.850     Found existing installation: setuptools 59.6.0
#13 3.851     Not uninstalling setuptools at /usr/lib/python3/dist-packages, outside environment /usr
---
#13 5.074   Downloading virtualenv-20.26.2-py3-none-any.whl (3.9 MB)
#13 5.132      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.9/3.9 MB 71.0 MB/s eta 0:00:00
#13 5.183 Collecting filelock<4,>=3.12.2
#13 5.186   Downloading filelock-3.14.0-py3-none-any.whl (12 kB)
#13 5.203 Collecting distlib<1,>=0.3.7
#13 5.207   Downloading distlib-0.3.8-py2.py3-none-any.whl (468 kB)
#13 5.213      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 468.9/468.9 KB 118.9 MB/s eta 0:00:00
#13 5.243 Collecting platformdirs<5,>=3.9.1
#13 5.246   Downloading platformdirs-4.2.2-py3-none-any.whl (18 kB)
#13 5.334 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#13 5.499 Successfully installed distlib-0.3.8 filelock-3.14.0 platformdirs-4.2.2 virtualenv-20.26.2
#13 DONE 5.6s

#14 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#14 DONE 0.0s
---
DirectMap4k:      233408 kB
DirectMap2M:     7106560 kB
DirectMap1G:    11534336 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint
    Finished `dev` profile [unoptimized] target(s) in 0.03s
##[endgroup]
downloading https://ci-artifacts.rust-lang.org/rustc-builds-alt/0de24a5177b1d49d6304f76f3ab159faaec134f9/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz
extracting /checkout/obj/build/cache/llvm-0de24a5177b1d49d6304f76f3ab159faaec134f9-true/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz to /checkout/obj/build/x86_64-unknown-linux-gnu/ci-llvm
---
fmt check
fmt: checked 5045 files
tidy check
tidy: Skipping binary file check, read-only filesystem
##[error]tidy error: /checkout/src/librustdoc/html/static/css/rustdoc.css:919: CSS files use tabs for indent
##[error]tidy error: /checkout/src/librustdoc/html/static/css/rustdoc.css:2063: CSS files use tabs for indent
##[error]tidy error: /checkout/src/librustdoc/html/static/css/rustdoc.css:2064: CSS files use tabs for indent
##[error]tidy error: /checkout/src/librustdoc/html/static/css/rustdoc.css:2065: CSS files use tabs for indent
removing old virtual environment
creating virtual environment at '/checkout/obj/build/venv' using 'python3.10'
Requirement already satisfied: pip in ./build/venv/lib/python3.10/site-packages (24.0)
some tidy checks failed
Build completed unsuccessfully in 0:01:08
  local time: Mon Jun 10 16:30:26 UTC 2024
  network time: Mon, 10 Jun 2024 16:30:26 GMT
```

</details>

GuillaumeGomez commented on src/librustdoc/html/static/css/rustdoc.css:
```diff
@@ -2059,6 +2060,10 @@ in src-script.js and main.js
 	}
 
+    .item-table>li>div {
+        max-width: unset !IMPORTANT;
```
Why is this needed?

BradMarr replied:
On mobile, the name stretches the whole way and the description comes below it. If I maintain it using only 40vw it will crop only halfway throughout the screen.

GuillaumeGomez replied:
I see. Then in this case, please remove this rule and move the other into the media query for desktop (which is at [this line](https://github.com/rust-lang/rust/blob/master/src/librustdoc/html/static/css/rustdoc.css#L2117)).

GuillaumeGomez commented on src/librustdoc/html/static/css/rustdoc.css:
```diff
@@ -898,8 +898,8 @@ because of the `[-]` element which would overlap with it. */
 */
-table,
-.item-table {
+.item-table,
```
This change can be removed.

rust-log-analyzer commented:
The job **`x86_64-gnu-tools`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/9452525772/job/26035948226) [(plain)](https://github.com/rust-lang/rust/commit/5fcad5f2738e295cf3a3c8e6629d60bec4e32a7c/checks/26035948226/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
.......... (120/124)
...F       (124/124)


/checkout/tests/rustdoc-gui/type-declation-overflow.goml type-declation-overflow... FAILED
[ERROR] `tests/rustdoc-gui/type-declation-overflow.goml` line 19: Error: Evaluation failed: The following errors happened (for selector `.item-table .struct`): [expected `684` for property `"offsetWidth"`, found `415`]: for command `assert-property: (".item-table .struct", {"offsetWidth": "684"})`
Error: ()
Build completed unsuccessfully in 0:03:27
  local time: Mon Jun 10 17:48:42 UTC 2024
  network time: Mon, 10 Jun 2024 17:48:42 GMT
```

</details>

BradMarr commented:
I'm confused on how to fix the check issues.

BradMarr commented:
> It says that in `tests/rustdoc-gui/type-declation-overflow.goml` at line 19, if expected a different value for the `offsetWidth` element property. Since you modified it, seems logical that it fails.

How should I go about fixing that?

GuillaumeGomez commented:
Update the mentioned file.

BradMarr commented:
> Update the mentioned file.

The mentioned file being the test or the css page?

GuillaumeGomez commented:
The test.

GuillaumeGomez commented on tests/rustdoc-gui/type-declation-overflow.goml:
```diff
@@ -14,10 +14,6 @@ assert-property: ("body", {"scrollWidth": "1100"})
 // However, since there is overflow in the type declaration, its scroll width is bigger.
 assert-property: ("pre.item-decl", {"scrollWidth": "1324"})
 
-// In the table-ish view on the module index, the name should not be wrapped more than necessary.
```
Not remove, update.

BradMarr commented:
How's this?

neo-zip commented:
LGTM!

GuillaumeGomez commented on src/librustdoc/html/static/css/rustdoc.css:
```diff
@@ -2127,6 +2128,10 @@ in src-script.js and main.js
 	}
+
+	.item-table > li > .item-name {
+		max-width: 40vw;
```
I just realized: why using the viewport width and not the parent element width?

BradMarr replied:
The parents width element is not static so I can't make the css `50%`

GuillaumeGomez replied:
The viewport width isn't either? Not sure to understand what you mean.

BradMarr replied:
When the item-name gets a width of 50% nothing changes. I'm assuming this is due to the fact that the parent doesn't have a predefined width. Instead I've set it to 40vw which seems to scale pretty nicely.

GuillaumeGomez replied:
That doesn't make any sense. The parent has a width. Also: you can resize the sidebar, changing the parent element width, but not the viewport width.

BradMarr replied:
Upon interacting with the parents width, using something like `max-width: 50%` on `.item-name`, no change occurs. I'm not sure exactly why this is the case but I'm guessing it has to do with how the parents width is defined.

GuillaumeGomez commented:
Also once you're done, please [squash](https://blog.guillaume-gomez.fr/Git-tips/1/2) your commits.

BradMarr commented:
What if we did something like this, where instead of a table it is a flex. It does look a little different but I don't mind the appearance.

rust-log-analyzer commented:
The job **`x86_64-gnu-tools`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/9508014711/job/26208568047) [(plain)](https://github.com/rust-lang/rust/commit/3fde88fef82ee4a82d76892db03878e2e191464d/checks/26208568047/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
.......... (120/124)
...F       (124/124)


/checkout/tests/rustdoc-gui/help-page.goml help-page... FAILED
[ERROR] `tests/rustdoc-gui/utils.goml` line 16
    from `tests/rustdoc-gui/help-page.goml` line 23
    from `tests/rustdoc-gui/help-page.goml` line 32: Error: Evaluation failed: The following errors happened: [localStorage item "rustdoc-theme" (of value "ayu") != "null"]: for command `assert-local-storage: {"rustdoc-theme": |theme|}`
Error: ()
Build completed unsuccessfully in 0:03:30
  local time: Fri Jun 14 00:04:50 UTC 2024
  network time: Fri, 14 Jun 2024 00:04:50 GMT
```

</details>

GuillaumeGomez commented:
The failure is "normal", not related to your changes. I restarted failing checks. If it still fails with the same error, just rebase.

GuillaumeGomez commented:
You messed up the git history. ^^'

BradMarr commented:
I think I'm a little bit out of my depth here, somebody else should probably take the reins.

GuillaumeGomez commented:
I'll take a look tomorrow.

GuillaumeGomez commented:
Closing in favor of https://github.com/rust-lang/rust/pull/127418.
